src/main/java/com/aws/greengrass/clientdevices/auth/metrics/MetricsEmitter.java

```diff
@@ -30,15 +29,31 @@ public MetricsEmitter(ScheduledExecutorService ses, ClientDeviceAuthMetrics metr
     }
 
     /**
-     * Start emitting metrics with no initial delay.
+     * Restart the metrics emitter and apply configurations.
+     *
+     * @param disableMetrics               Enable metrics flag
+     * @param periodicAggregateIntervalSec Periodic aggregate interval in seconds
      */
-    public void start() {
+    public void restart(boolean disableMetrics, int periodicAggregateIntervalSec) {
+        if (disableMetrics) {
+            return;
+        }
+
+        stop();
+        start(periodicAggregateIntervalSec);
+    }
+
+    /**
+     * Start emitting metrics with no initial delay unless specified otherwise by the Metrics Configuration.
```

Comment:
> unless specified otherwise by the Metrics Configuration.

this isn't true though? initial delay is always zero

Reply:
Forgot to edit the comment, but the otherwise part was related to starting it, not the initial delay. Will go back and fix it